lib/spec.js

```diff
@@ -20,8 +20,22 @@ function getReleaseNumber(release) {
 
 function getVersionNumber({ spec, version }) {
   const replaceHyphens = getValueFromSpec(spec, 'replaceHyphens', false);
+  const allowedReplacements = ['~', '_', '+'];
+  let newVersion;
 
-  return replaceHyphens ? version.replace(/-/g, '~') : version;
+  if (replaceHyphens === true) {
+    // Stay backwards compatible. Previous version was true -> tilde.
+    newVersion = version.replace(/-/g, '~');
+  } else if (typeof replaceHyphens === 'string') {
+    if (allowedReplacements.includes(replaceHyphens)) {
+      newVersion = version.replace(/-/g, replaceHyphens);
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn(`replaceHyphens was given a forbidden string, so no replacement was done. Please use one of ${allowedReplacements.join('')}`);
```

Comment:
My preference would be to throw this as an `Error` here instead of (almost) silently ignoring it